RevenueCatUI/CustomerCenter/Views/RestorePurchasesAlert.swift

```diff
@@ -56,7 +58,11 @@ struct RestorePurchasesAlert: ViewModifier {
                                     """),
                         primaryButton: .default(Text("Check past purchases"), action: {
                             Task {
+                                isRestoring = true
                                 let alertType = await self.customerCenterViewModel.performRestore()
+                                // Simulate a delay to ensure the progress view is shown for at least one second
+                                await Task.sleep(seconds: 1)
```

Comment:
Without this, the text flashes. I figured it was actually not too bad, but maybe there's a better way to guarantee the progress view is displayed for a bit, or at least it doesn't flash?

Reply:
🤔  maybe we can introduce an animation that has a delay, I think we've done it before (or I'm thinking about a personal app 😅 )

Reply:
Some chatGPT suggestions, but it's pretty much what I was thinking about

```swift
if showProgressIndicator {
    ProgressView()
        .transition(.opacity)
}

// and then when you want to show it: 
DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
    if isLoading {
        withAnimation {
            showProgressIndicator = true
        }
    }
}
```

Reply:
I might be wrong, but doesn't this now add a second after however long `performRestore()` took? I deduced that from [this blog](https://www.avanderlee.com/swift/async-let-asynchronous-functions-in-parallel/). I think running it in parallel with `performRestore()` might be nicer? What do you think?

Reply:
Oh good points. The animation could be the right approach to this, but if that doesn't work I will try parallelising. Thanks!